### PR TITLE
Feature/enhanced cc

### PR DESCRIPTION
**Pull request by fu**
Enhanced ChemicalComposition so
a) unimod xml is parsed only once and not every time the class is initialised 
b) _mass also can be used with e.g. TMT6Plex that contains 13C

### CODE REVIEW

File: pyqms/chemical_composition.py [hunk 1]
```diff
@@ -289,7 +301,7 @@ def _parse_sequence_unimod_style( self, sequence ):
             # print( self , 'peptide only')
             # print( 'Unimod:', unimod, unimod[:end] , )
             # Full addition
-            # print( unimodcomposition , '<<<<<<')
+            print( unimodcomposition , '<<<<<<', ChemicalComposition._unimod_parser)
```
[the remark this answers is not in the document]
Comment (fu) [reply pyqms/chemical_composition.py]: good catch .. shouldn't be there .. removed now

**Comment by fu** [comment]
Added Test that checks for TMT6plex mass ... Thanks the review 👍

File: pyqms/chemical_composition.py [hunk 2]
```diff
@@ -531,8 +543,14 @@ def _mass(self, cc=None):
         else:
             cc_mass_dict = cc
         for element, count in cc_mass_dict.items():
-            mass += count * self.isotopic_distributions[element][0][0]
-
+            if element not in self.isotopic_distributions.keys():
+                match = re.search('(?P<isotope>[0-9]*)(?P<element>[A-Z][a-z]*$)', element)
+                for _emass, _distribution in self.isotopic_distributions[match.group('element')]:
+                    if int(round(mass)) == int(match.group('isotope')):
```
[the remark this answers is not in the document]
Comment (fu) [reply pyqms/chemical_composition.py]: no worries, fixed it and added a test. Hope that is good now :)

File: pyqms/chemical_composition.py [hunk 3]
```diff
@@ -546,8 +546,9 @@ def _mass(self, cc=None):
             if element not in self.isotopic_distributions.keys():
                 match = re.search('(?P<isotope>[0-9]*)(?P<element>[A-Z][a-z]*$)', element)
                 for _emass, _distribution in self.isotopic_distributions[match.group('element')]:
-                    if int(round(mass)) == int(match.group('isotope')):
+                    if int(round(_emass)) == int(match.group('isotope')):
```
Comment (MKoesters) [inline pyqms/chemical_composition.py]: Ah, makes sense now :)

**Comment by StSchulze** [comment]
This might be slightly off topic but we should work on merging ChemicalComposition from Ursgal and pyQms ... ideally before adding more to either of the two.

But since I had assigned that task to myself some time ago, I just go with: looks good, approved :P

**Pull request merged**